server.js

```diff
@@ -1,30 +1,203 @@
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+
+// Defines the port the app will run on.
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Start ROUTE on defalt port 8080.
+
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+   res.send({
+    responseMassage: "Hello Everybody and welcome!",
+    guide: "These are the routes for this TED Talk API!",
+    Endpoints:"",
+    Routes:[{
+        "/allTedTalks": "All TED Talks",
+        "/speakers": "All TED Talk speakers, and how many",
+        "/speakers/:speaker/talks": "TED Talks from a Requested speaker, and how many", 
+        "/events": "All TED Talk events",
+        "/tedTalk/:id": "search for a TED Talk with specific Id",
+        "/tedTalks/mostViewedTedTalk": "How many views has the most popular talk?",
+        "/randomTedTalk": "Press enter and get a random TED Talk"
+      }],
+    Querys: [{
+      "/tedTalks": "Filter by title, speaker and/or event, amoung a selected amount (500) of talks ex. /tedTalks?speaker=al gore&event=ted2006",
+    }]
+  });
+});
+
+// Route 1 - collection of results (array of elements)
+app.get("/allTedTalks", (req, res) => {
+  res.status(200).json(tedTalkData); 
+});
+
+// Route 2 all speakers list
+
+app.get("/speakers", (req, res) => {
+  const allSpeakers = tedTalkData.map((talk) => talk.speaker);
+  // New set to remove all dublicates
+  const allSpeakersSet = Array.from(new Set(allSpeakers));
+  if (allSpeakersSet){
+    res.status(200).json({
+      success: true,
+      message: "All TED Talk spekers",
+      'Total amount of categories': allSpeakersSet.length,
+      body: {
+          tedTalkData: allSpeakersSet
+      }
+    })
+  }
+});
+
+// Route 3 talks by requested speakers list
+// ex /speaker/hans rosling/talks
+
+app.get("/speakers/:speaker/talks", (req, res) => {
+   const talksBySpeaker = tedTalkData.filter((talkSpeaker) => {
+    return talkSpeaker.speaker.toLowerCase() === req.params.speaker.toLowerCase();
+  })
+  if (talksBySpeaker) {
+    res.status(200).json({
+      success: true,
+      message: "All TED Talks by requested speker",
+      'Amount of talks': talksBySpeaker.length,
+      body: {
+        tedTalkData: talksBySpeaker
+      }
+    })
+  }
+});
+
+// Route 4 All TED Talk Events list
+// ex /events
+
+app.get("/events", (req, res) => {
+  const allEvents = tedTalkData.map((talkEvent) => talkEvent.event);
+  // New set to remove all dublicates
+  const allEventsSet = Array.from(new Set(allEvents));
+
```

Comment:
Very impressive, we haven't learned about `new Set()` in this course but this is a very good use case :)